### PR TITLE
Add Codecov reporting.

### PR DESCRIPTION
**Pull request by jezdez**
This also restructures the way the Circle config calls the tests a bit by extending the bin/run script.

### Pull Request checklist

- **Documentation**: If this feature has or requires documentation, the relevant docs have been updated.
- [x] **Changelog**: This PR updates the [changelog](https://github.com/iodide-project/iodide/blob/master/CHANGELOG.md) with any user-visible changes.
- **Tests**: This PR includes thorough tests or an explanation of why it does not

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # (Unreleased; add upcoming change notes here)
 
+# unreleased
+
```
Comment (wlach) [inline CHANGELOG.md]: There is an unreleased header just above, no need to add this :)

Comment (wlach) [inline CHANGELOG.md]:
```suggestion
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
 # (Unreleased; add upcoming change notes here)
 
+# unreleased
```
Comment (wlach) [inline CHANGELOG.md]:
```suggestion
```

File: CHANGELOG.md (hunk 1 above)
Comment (jezdez) [reply CHANGELOG.md]: D'oh!

**Pull request merged**